torch/cuda/amp/autocast_mode.py

```diff
@@ -5,7 +5,7 @@
 try:
     import numpy as np
 except ModuleNotFoundError:
-    np = None
+    np = None  # type: ignore[assignment]
```

Comment:
This is minor, but: the normal way to do this is:
```
try:
    import numpy as np
    HAS_NUMPY = True
except ModuleNotFoundError:
    HAS_NUMPY = False
```
and change the code further down to match.